### PR TITLE
[ao] Updated detector observer insert args to be vars not strings

### PR DESCRIPTION
**Pull request by vspenubarthi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81369
* __->__ #81382

Summary: Before for the detectors, the
determine_observer_insert_points() function for all of them would have
hard coded strings as the keys for the dictionary that would be returned
to the ModelReport instance, and those same hard-coded keys would be
used to actually extract information from them. Since all detectors used
the same string keys, these were just made default variables at the top
of the detector.py file, and all detectors just used those. The same
ones are imported and now used in ModelReport file as well. This way,
there is less of a chance of an error because of incorrectly typed
strings.

The test plan primarily tests the ModelReport class because this uses
the same new vars as well for the strings and is the primary one calling
each of the detector instances' determine_observer_insert_points()

Test Plan: python test/test_quantization.py TestFxModelReportClass

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: torch/ao/quantization/fx/_model_report/detector.py [hunk 1]
```diff
@@ -11,6 +11,12 @@
 from torch.ao.quantization.qconfig import QConfig
 from torch.ao.quantization.quantize import is_activation_post_process
 
+# Names for observer insert keys
+DEFAULT_TARGET_NODE_KEY = "target_node"
+DEFAULT_OBSERVER_INSERT_KEY = "observer_to_insert"
+DEFAULT_POST_KEY = "is_post_observer"
+DEFAULT_OBS_ARGS_KEY = "observer_args"
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_model_report/detector.py:18]: we can remove `DEFAULT` probably since we won't be updating them?

Comment (jerryzh168) [reply torch/ao/quantization/fx/_model_report/detector.py:18]: we can prefix with something like `DETECTOR_` maybe

Comment (vspenubarthi) [reply torch/ao/quantization/fx/_model_report/detector.py:18]: changed to prefix detector instead, thanks for the suggestion

File: torch/ao/quantization/fx/_model_report/detector.py [hunk 2]
```diff
@@ -11,6 +11,12 @@
 from torch.ao.quantization.qconfig import QConfig
 from torch.ao.quantization.quantize import is_activation_post_process
 
+# Names for observer insert keys
+DETECTOR_TARGET_NODE_KEY = "target_node"
+DETECTOR_OBSERVER_INSERT_KEY = "observer_to_insert"
+DETECTOR_POST_KEY = "is_post_observer"
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_model_report/detector.py:17]: nit: maybe just spell these out to make it clearer, e.g.
'DETECTOR_OBS_TO_INSERT_KEY' and 'DETECTOR_IS_POST_OBS_KEY'

**Review by jerryzh168 — APPROVED** [review]

**Pull request closed without merging**